ls1intum/Artemis
# Feature/programming exercise/move preliminary check to client

### Checklist

### Motivation and Context



Follow up to https://github.com/ls1intum/Artemis/pull/869.
Move the preliminary rendering to the client.

### Description


The preliminary string is shown if:
- The participation result is of a programming exercise student participation AND
- The buildAndTestAfterDueDate of the programming exercise is defined and set in the future.

Note: I did not include handling for result strings that already container the 'preliminary' string because it was generated on the server to reduce the complexity of the implementation.
I think we don't have to handle this case as there shouldn't be real student results with the preliminary result string.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration > Programming Exercise
3. Set the buildAndTestAfterDueDate in future => preliminary should be shown on student results

####

1. Log in to Artemis
2. Navigate to Course Administration > Programming Exercise
3. Remove the buildAndTestAfterDueDate => preliminary should not be shown on student results

### Screenshots



![image](https://user-images.githubusercontent.com/28230611/66483781-0c19d100-eaa6-11e9-8557-4ba8d4934fdc.png)
